lib/rubocop/cop/style/unpack_first.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for accessing the first element of *unpack* which can
+      # be replaced with the shorter method *unpack1*.
+      #
+      # @example
+      #
+      #   # bad
+      #   'foo'.unpack('h*').first
+      #   'foo'.unpack('h*')[0]
+      #   'foo'.unpack('h*').slice(0)
+      #   'foo'.unpack('h*').at(0)
+      #   'foo'.unpack('h*').take(1)
+      #
+      #   # good
+      #   'foo'.unpack1('h*')
+      #
+      class UnpackFirst < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.4
+
+        MSG = 'Use `String#unpack1(%<format>s)` instead of '\
+          '`String#unpack(%<format>s)%<method>s`.'.freeze
+
+        def_node_matcher :unpack_and_first_element?, <<-PATTERN
```

Comment:
Thoughts for later: I wonder if we could capture this into a reusable form to make it easier for other cops to also identify that they are grabbing the first element from an array. `Performance/Sample` also looks for grabbing the first element, but is missing checks for `at` and `take`.